website/integrations: aws classic: migrate to entitlements (#21674)
goauthentik/authentik: website/integrations/cloud-providers/aws-classic/index.mdx

## authentik configuration

> To support the integration of AWS with authentik via the Classic IAM method, you need to create two property mappings, an application/provider pair, and application entitlements for the AWS roles users should be able to assume.

Reviewer: ```suggestion
To support the integration of AWS with authentik via the Classic IAM method, you need to create two property mappings, an application/provider pair, and application entitlements for the AWS roles that users can assume.
```

        return "arn:aws:iam::<aws_account_ID>:role/authentik-admin,arn:aws:iam::<aws_account_ID>:saml-provider/<provider_name>"
        ```

        **For role assignment based on application entitlements** (recommended):

        ```python
        ACCOUNT_ID = "<aws_account_ID>"
        PROVIDER_NAME = "<provider_name>"
        roles = [
            entitlement.name
            for entitlement in request.user.app_entitlements(provider.application)
        ]

        return [
            f"arn:aws:iam::{ACCOUNT_ID}:role/{r},arn:aws:iam::{ACCOUNT_ID}:saml-provider/{PROVIDER_NAME}"
            for r in roles
        ]
        ```

        :::tip

>         With the entitlement-based approach, create application entitlements with names like `authentik-admin` and `authentik-read-only`, then bind the appropriate users or groups to those entitlements. Create these entitlements after you create the application, as described later in this guide. Users receive the corresponding AWS roles when they access the AWS application.

Reviewer: ```suggestion
        With the entitlement-based approach, create application entitlements with names like `AWS:authentik-admin` and `AWS:authentik-read-only`, then bind the appropriate users or groups to those entitlements. Create these entitlements after you create the application, as described later in this guide. Users receive the corresponding AWS roles when they access the AWS application.
```

We should either keep the `AWS:` or rename the groups. Otherwise they too closely match the default authentik group names.
